### PR TITLE
Format Integers and decimals of balances on home screen

### PR DESCRIPTION
**Pull request by youngkidwarrior**
@0xBigBoss not happy with the approach here, but this would save us from having to write custom logic for decimal and integer per token

Care to give some input. Commented a test out that would no longer be relevant.

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/415?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#416** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/416?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#415** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/415?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#414** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/414?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#413** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/413?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#412** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/412?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by 0xBigBoss** [comment]
i'm not really grokking what this is supposed to solve

**Comment by youngkidwarrior** [comment]
It removes any trailing zeros from the decimals

So `formatAmount(1024.44000, 4,4) = `1024.44`

Its really good for when we show different USD prices and token balances from the same component.

e.g.

Send price = 0.00033 USD [API Link](api.coingecko.com/api/v3/coins/markets?ids=send-token&vs_currency=usd)
Ethereum price = 1024.24 USD [API Link](api.coingecko.com/api/v3/coins/markets?ids=ethereum&vs_currency=usd)

these changes makes it easy to format both

**Comment by 0xBigBoss** [comment]
> It removes any trailing zeros from the decimals
> 
> So `formatAmount(1024.44000, 4,4) = `1024.44`
> 
> Its really good for when we show different USD prices and token balances from the same component.
> 
> e.g.
> 
> Send price = 0.00033 USD [API Link](api.coingecko.com/api/v3/coins/markets?ids=send-token&vs_currency=usd) Ethereum price = 1024.24 USD [API Link](api.coingecko.com/api/v3/coins/markets?ids=ethereum&vs_currency=usd)
> 
> these changes makes it easy to format both
> 
> The current way would show Ethereum at 1024.44000.
> 
> This might be better adjusted to have a function of its own

ok, this makes sense. yeah let's add a new function, (potentially just copy-pasta-ing this one), and allow for configuring these new formatting options with a better API. Or "carefully" update the API of this one and add a sane default option for this new feature.

Or just make a new one-off function that does what your aiming to do

**Review by youngkidwarrior — COMMENTED** [review]
@0xBigBoss I noticed that I really only need to handle trailing zeros with the coingecko data. So I wrote an internal function for that component instead, avoiding modifying the `formatAmount` utility function

File: packages/app/features/home/TokenDetails.tsx [hunk 1]
```diff
@@ -89,6 +95,8 @@ export const TokenDetailsMarketData = ({ coin }: { coin: coins[number] }) => {
       </XStack>
     )
 
+  const formatPrice = (price: number) => price.toString().slice(0, 7)
```
Comment (0xBigBoss) [inline packages/app/features/home/TokenDetails.tsx:98]: confused what this is achieving? only take first 7 characters? can we add a why comment?

**Review by 0xBigBoss — APPROVED** [review]

File: packages/app/features/home/TokenDetails.tsx (hunk 1 above)
Comment (youngkidwarrior) [reply packages/app/features/home/TokenDetails.tsx:98]: Yea coingecko formats the price for us, so this just makes sure that the text fits in the component. Its simple and should handle fine for our current selection of coins

Comment (youngkidwarrior) [reply packages/app/features/home/TokenDetails.tsx:98]: Ill add a note

**Pull request merged**